src/Microsoft.DotNet.XHarness.CLI/Commands/Apple/AppleOrchestrator.cs

```diff
@@ -0,0 +1,366 @@
+                }
+                catch (Exception e)
+                {
+                    var message = new StringBuilder().AppendLine("Application run failed:");
+
+                    if (_errorKnowledgeBase.IsKnownTestIssue(_mainLog, out var failureMessage))
+                    {
+                        message.Append(failureMessage.Value.HumanMessage);
+                        if (failureMessage.Value.IssueLink != null)
+                        {
+                            message.AppendLine($" Find more information at {failureMessage.Value.IssueLink}");
+                        }
+                    }
+                    else
+                    {
+                        message.AppendLine(e.ToString());
+                    }
+
+                    _logger.LogError(message.ToString());
+
+                    return ExitCode.APP_LAUNCH_FAILURE;
+                }
+            }
+
+            try
+            {
+                (device, companionDevice) = await _deviceFinder.FindDevice(target, arguments.DeviceName, _mainLog);
+
+                _logger.LogInformation($"Found {(target.Platform.IsSimulator() ? "simulator" : "physical")} device '{device.Name}'");
+
+                if (companionDevice != null)
+                {
+                    _logger.LogInformation($"Found companion {(target.Platform.IsSimulator() ? "simulator" : "physical")} device '{companionDevice.Name}'");
+                }
+
+                if (target.Platform.IsSimulator() && arguments.ResetSimulator)
+                {
+                    var simulator = (ISimulatorDevice)device;
+
+                    _logger.LogInformation($"Reseting simulator '{device.Name}'");
+                    await simulator.PrepareSimulator(_mainLog, appBundleInfo.BundleIdentifier);
+
+                    if (companionDevice != null)
+                    {
+                        _logger.LogInformation($"Reseting companion simulator '{companionDevice.Name}'");
+                        var companionSimulator = (ISimulatorDevice)companionDevice;
+                        await companionSimulator.PrepareSimulator(_mainLog, appBundleInfo.BundleIdentifier);
+                    }
+
+                    _logger.LogInformation("Simulator reset finished");
+                }
+            }
+            catch (NoDeviceFoundException e)
+            {
+                _logger.LogError(e.Message);
+                return ExitCode.DEVICE_NOT_FOUND;
+            }
+
+            exitCode = await InstallApp(appBundleInfo, device, target, cancellationToken);
+
+            if (exitCode != ExitCode.SUCCESS)
+            {
+                if (!target.Platform.IsSimulator())
+                {
+                    _logger.LogInformation($"Cleaning up the failed installation from '{device.Name}'");
+                    await UninstallApp(appBundleInfo, device, new CancellationToken());
```

Comment:
why not uninstalling for simulators?

Reply:
mlaunch just cannot do it, I am not sure what is the reason behind that